packages/backend-core/src/users/db.ts

```diff
@@ -372,21 +372,26 @@ export class UserDB {
     }
 
     const account = await accountSdk.getAccountByTenantId(tenantId)
+    const isSSOEnforced = await UserDB.features.isSSOEnforced()
+
     return UserDB.quotas.addUsers(
       newUsers.length,
       newCreators.length,
       async () => {
-        // create the promises array that will be called by bulkDocs
         newUsers.forEach((user: any) => {
+          if (isSSOEnforced && user.password) {
```

Comment:
Should this not be just checking `if (isSSOEnforced)`, to get sure we don't have empty strings or other kind of value on `password`?